src/components/button/config/demo.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright IBM Corp. 2018, 2019
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/**
+ * Configuration for the Button component
+ * @module button/demo
+ */
+import startCase from 'lodash/startCase';
+import { generate } from './generate';
+
+/**
+ * Creates a button template configuration for a variant
+ * @returns {globalTypedefs.fractalDemo} fractal demo object
+ */
+const variantButtons = function() {
+  const variants = {};
+
+  Object.keys(this.selectors.variants).forEach(variant => {
+    variants[variant] = {
+      name: variant,
+      label: startCase(variant),
+      context: generate.apply(this, [
+        {
+          variant,
+        },
+      ]),
+    };
+  });
+
+  return variants;
+};
+
+/**
+ * Modifier: Anchor
```

Comment:
Personally I don't find "anchor" to be a "modifier" as there is a lot that goes in to making a "button" a link.  It is not on the same level as making a button "small". 

That was something we worked out extensively to define the difference between the terms `variant`, `modifier`, `state`, and `scenario`.

Reply:
Assuming being an "anchor" means we set different `element` and an additional `aria` role... playing along with Carbon React for a second... how would you classify this?

Can you share those definitions?

Reply:
* **variant** - larger styling/functionality differences.  You can't mix variants on the same instance. Examples include:
   - primary, secondary, etc for buttons
   - required action, transactional, passive modals
* **modifier** - small change that can be applied to any/multiple variants. However modifiers of the same "category" can not be applied to the same instance.  Examples include:
  - Size: such as trying to add compact/comfy (small/large) classes to the same instance
  - row differentiators for tables (alternating colors vs borders)
  - direction of a tooltip in relation to it's trigger
* **state** - change caused by the user interacting with an element/component
* **scenario** - changes the HTML structure/content model  (e.g. button with an icon, data table with a toolbar, tooltip with a heading, etc)

This terminology we use in the selectors object, config definitions, etc.  For demo description specifics, we also have a `test` category which is used to describe a demo that is created specifically to test different setups/bugs and does not define a feature of a component.

Reply:
This info should be documented somewhere.

Reply:
I know... I thought I had it on our site somewhere but couldn't find it.  📝